src/Model/EloquentQueryBuilder.php

```diff
@@ -24,34 +24,51 @@ class EloquentQueryBuilder extends Builder
     protected $model;
 
     /**
+     * Store a runtime cache so that there are never any duplicated queries.
+     *
+     * @var array
+     */
+    protected static $modelCache = [];
```

Comment:
Lets just call it cache - you know how I feel about multiple syllables and words.